src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -142,6 +142,39 @@ public void Commit()
         #region Helpers
 
         private void AddRuleset()
+        {
+            TrySetNonConditionalRuleSet();
+            SetConditionalRuleSets();
+        }
+
+        /// <summary>
+        /// Try to set a ruleset without specifying a configuration (e.g. Debug / Release).
+        /// No changes are made if the project already has a ruleset.
+        /// </summary>
+        private void TrySetNonConditionalRuleSet()
+        {
+            var hasUserSpecifiedRuleSet = propertyInformationMap.Keys.Any(x => !IsDefaultMicrosoftRuleSet(x.Value as string));
+
+            if (hasUserSpecifiedRuleSet)
+            {
+                // We could've done this for all projects that have configurations that point to the same ruleset but we don't want to create noise for existing users.
```

Comment:
I don't understand this comment. We could have done what?

Reply:
Set a single non-conditional ruleset